### PR TITLE
release-22.2: allocator: expose LeaseRebalanceThreshold as a cluster setting

### PR DESCRIPTION
**Pull request by wenyihu6**
Backport 1/1 commits from #105924.

/cc @cockroachdb/release

---

This patch adds a new cluster setting --
`kv.allocator.lease_rebalance_threshold`. This setting controls the minimum
fraction away from the mean a store's lease count can be before it is considered
for lease-transfers. The default setting is 0.05.

Fixes: https://github.com/cockroachdb/cockroach/issues/105909

Release Note (allocator): cluster setting
`kv.allocator.lease_rebalance_threshold` can now be used to control the minimum 
fraction away from the mean a store's lease count before it is considered for
lease-transfers. The default setting is 0.05.

---

Release justification: low risk to the existing functionality

### CODE REVIEW

**Comment by wenyihu6** [comment]
Same comment from the previous PR - non-trivial backport to the 22.2 branch due to the absence of the cluster setting field on the allocator. Happy to close this PR if the change is considered substantial.

Would appreciate a second pair of eyes on adding the cluster setting field to the allocator. Most of the changes were made by just observing the existing pattern observed on the master branch.

**Review by kvoli — APPROVED** [review]
:lgtm: 

Are you going to squash the first commit into the second before merging?

> Reviewed 6 of 6 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/106186#-:-N_-2ag7CH1naMr90vSi:bbtfew2)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @wenyihu6)

**Pull request merged**
